File: python/sky-sync.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import boto3
import subprocess

ec = boto3.resource('ec2')

#instances = ec.instances.filter(Filters=[{'Name': 'instance-state-name', 'Values': ['running']}])
instances = ec.instances.filter(Filters=[{'Name': 'tag-key', 'Values': ['Name']}])

aws_nodes = []

for instance in instances:
    hostname = [tag for tag in instance.tags if tag['Key'] == 'Name']
    aws_nodes.append(hostname[0]['Value'].split()[0])

print(len(aws_nodes))
#print(aws_nodes)

chef_nodes = subprocess.check_output(["knife", "node", "list"]).split()
print(len(chef_nodes))

for node in chef_nodes:
    if node not in aws_nodes:
        print(node)

File: ruby/sky-sync.rb
#!/usr/bin/env ruby


require 'aws-sdk'
require 'chef-api'


ec2 = Aws::EC2::Resource.new(region: 'us-east-1')


# Set empty list for active AWS Hosts
#activeHosts = Array.new

# Get a list of all current AWS Hosts
#f = [ { name: 'instance-state-name', values: ['running'] } ]
#ec2.instances().each do |i|
#	name = i.tags.select{|tag| tag.key == 'Name'}
#	activeHosts.push("#{name[0].value.split[0]}")
#	
#end

#puts activeHosts.display


# Set empty list for active Chef Nodes
#chefNodes = Array.new

